bsi_zoo/estimators.py

```diff
@@ -63,6 +68,76 @@ def _solve_reweighted_lasso(
     return x
 
 
+class Solver(BaseEstimator, ClassifierMixin):
+    def __init__(self, solver, alpha=None, cov=None):
+        self.solver = solver
+        self.alpha = alpha
+        self.cov = cov
+
+    def fit(self, L, y):
+        self.L_ = L
+        self.y_ = y
+
+        return self
+
+    def predict(self, y):
+        if self.cov is None:
+            self.coef_ = self.solver(self.L_, y, alpha=self.alpha)
+        else:
+            self.coef_ = self.solver(self.L_, y, self.cov, alpha=self.alpha)
+
+        return self.coef_
+
+
+class Spatial_CV_Solver(BaseEstimator, ClassifierMixin):
+    def __init__(self, solver, alpha=None, alphas = np.linspace(1.4, 0.1, 20), cov=None):
+        self.solver = solver
+        self.alpha = alpha
+        self.alphas = alphas
+        self.cov = cov
+
+    def fit(self, L, y):
+        self.L_ = L
+        self.y_ = y
+        clf = GridSearchCV(
+            estimator=Solver(self.solver, cov=self.cov),
+            param_grid=dict(alpha=self.alphas),
+            scoring="neg_mean_squared_error",
+            cv=5,
+            n_jobs=-1,
+        )
+        clf.fit(L, y)
+        self.alpha = clf.best_estimator_.alpha
+        return self
+
+    def predict(self, y):
+        if self.cov is None:
+            self.coef_ = self.solver(self.L_, y, alpha=self.alpha)
+        else:
+            self.coef_ = self.solver(self.L_, y, self.cov, alpha=self.alpha)
+
+        return self.coef_
+
+
+def run_spatial_cv(solver, L, y, cov=None, n_orient=1, **extra_params):
```

Comment:
This function is now redundant with the class. Either the class uses this function or we remove it